core/src/main/scala/testz/data.scala

```diff
@@ -93,3 +93,24 @@ object Result {
       new Fail(first.asInstanceOf[Fail].failures ++ second.asInstanceOf[Fail].failures)
     }
 }
+
+final class TestOutput(
+  val failed: Boolean,
+  val print: () => Unit
+)
+
+object TestOutput {
+  def combine(fst: TestOutput, snd: TestOutput) =
+    new TestOutput(
+      fst.failed || snd.failed,
+      { () => fst.print(); snd.print() }
+    )
+
+  def combineAll1(output1: TestOutput, outputs: TestOutput*) = {
```

Comment:
arguably this is monoidal and not just semigroupal, but I'm sure you considered that.

Reply:
Exactly, people shouldn't have empty test suites.

Reply:
but I can do it with scalatest!